src/ukify/ukify.py

```diff
@@ -848,16 +857,18 @@ def call_systemd_measure(uki: UKI, opts: UkifyConfig, profile_start: int = 0) ->
                 *(f'--bank={bank}' for bank in banks),
             ]
 
-            for priv_key, pub_key, group in key_path_groups(opts):
+            for priv_key, pub_key, cert, group in key_path_groups(opts):
                 extra = [f'--private-key={priv_key}']
                 if opts.signing_engine is not None:
-                    assert pub_key
+                    assert cert
                     extra += [f'--private-key-source=engine:{opts.signing_engine}']
-                    extra += [f'--certificate={pub_key}']
+                    extra += [f'--certificate={cert}']
                 elif opts.signing_provider is not None:
-                    assert pub_key
+                    assert cert
                     extra += [f'--private-key-source=provider:{opts.signing_provider}']
-                    extra += [f'--certificate={pub_key}']
+                    extra += [f'--certificate={cert}']
+                elif cert:
+                    extra += [f'--certificate={cert}']
                 elif pub_key:
                     extra += [f'--public-key={pub_key}']
```

Comment:
This also looks like a compat break. This should always use pub_key if it is set and otherwise cert if it is set.

Reply:
ok I've added some backwared compat glue